### PR TITLE
Graphql/private

### PR DESCRIPTION
**Pull request by JanuszJakubiec**

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/123446) / small_tests / 497ef24601eaad43d5fc506d83cdd4c4ab2bc05a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3652/123446/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3652/123446/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-05-23_06.24.19/lib.mongooseim.logs/run.2022-05-23_06.24.20/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/123481) / small_tests / 6bba19ff3b758bc3145d448e5c539b97d6625329
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3652/123481/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3652/123481/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-05-23_06.27.34/lib.mongooseim.logs/run.2022-05-23_06.27.35/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/123517) / small_tests / a475d155725b6b0a11f9bc24c763cc29ced855a9
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3652/123517/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3652/123517/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-05-23_06.29.33/lib.mongooseim.logs/run.2022-05-23_06.29.34/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/123495) / small_tests / 68e860decbe765c81c54fd470d0e2753e524555e
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3652/123495/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3652/123495/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-05-23_06.28.52/lib.mongooseim.logs/run.2022-05-23_06.28.53/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/123548) / small_tests / ebd71ee1109d1c858daa2a4057da4f2b705eba02
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3652/123548/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3652/123548/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-05-23_06.36.19/lib.mongooseim.logs/run.2022-05-23_06.36.19/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/123564) / small_tests / 55efd1f3e16959456308c4f7bfa7a00d2e0e5a0d
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3652/123564/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3652/123564/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-05-23_08.54.39/lib.mongooseim.logs/run.2022-05-23_08.54.40/suite.log.html)

File: src/graphql/user/mongoose_graphql_private_user_mutation.erl [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+-export([execute/4]).
+
+-ignore_xref([execute/4]).
+
+-include("../mongoose_graphql_types.hrl").
+
+-import(mongoose_graphql_helper, [make_error/2, format_result/2]).
+
+execute(#{user := CallerJID}, _Obj, <<"setPrivate">>,
+        #{<<"privateString">> := Element}) ->
```
Comment (chrzaszcz) [inline src/graphql/user/mongoose_graphql_private_user_mutation.erl:13]: This would fit into one line

File: priv/graphql/schemas/user/private.gql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+"""
+Allow user to set own private
+"""
+type PrivateUserMutation @protected {
+    setPrivate(privateString: String!): String
```
Comment (chrzaszcz) [inline priv/graphql/schemas/user/private.gql:5]: This looks like it could be any string, while it needs to be valid XML. Maybe `elementString`?

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/123814) / small_tests / 462607c2be49f0f754af5df7867b726db29a8802
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3652/123814/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3652/123814/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-05-24_09.04.07/lib.mongooseim.logs/run.2022-05-24_09.04.08/suite.log.html)

File: priv/graphql/schemas/admin/private.gql [hunk 3]
```diff
@@ -0,0 +1,13 @@
+"""
+Allow admin to set user's private
+"""
+type PrivateAdminMutation @protected {
+    setPrivate(user: JID!, elementString: String!): String
```
Comment (Premwoik) [inline priv/graphql/schemas/admin/private.gql:5]: I would add documentation here because other commands have it :)

**Review by Premwoik — APPROVED** [review]
Looks good :)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/123988) / small_tests / 45ac0048cd8e31ea8701d02cab6fe084920ab5dd
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3652/123988/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3652/123988/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-05-25_06.09.13/lib.mongooseim.logs/run.2022-05-25_06.09.13/suite.log.html)

**Review by chrzaszcz — APPROVED** [review]
Looks good 👍

**Pull request merged**